src/compiler/visitorPublic.ts

```diff
@@ -390,916 +1158,12 @@ namespace ts {
             return node;
         }
 
-        const factory = context.factory;
-        switch (kind) {
-            // Names
-
-            case SyntaxKind.Identifier:
-                Debug.type<Identifier>(node);

```

Comment:
This is maybe a silly question, but are we sure the perf boost here isn't only because the new code never calls `Debug.type`?

Reply:
`Debug.type()` had almost no impact on perf when I added it. It's a noop, so inlining pretty much results in elimination. It definitely wouldn't account for a 3% difference in emit time.

Reply:
Fair enough, just checking!